src/Identity/EntityFrameworkCore/test/EF.Test/VersionTwoSchemaTest.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Identity.EntityFrameworkCore.Test;
+
+public class VersionTwoSchemaTest : IClassFixture<ScratchDatabaseFixture>
+{
+    private readonly ApplicationBuilder _builder;
+
+    public VersionTwoSchemaTest(ScratchDatabaseFixture fixture)
+    {
+        var services = new ServiceCollection();
+
+        services
+            .AddSingleton<IConfiguration>(new ConfigurationBuilder().Build())
+            .AddDbContext<VersionTwoDbContext>(o =>
+                o.UseSqlite(fixture.Connection)
+                    .ConfigureWarnings(b => b.Log(CoreEventId.ManyServiceProvidersCreatedWarning)))
+            .AddIdentity<IdentityUser, IdentityRole>(o =>
+            {
+                // MaxKeyLength does not need to be set in version 2
+                o.Stores.SchemaVersion = IdentitySchemaVersions.Version2;
```

Comment:
Will this be part of the new scaffolding since it needs to be opted into? Will `AddDefaultIdentity` continue to set the `MaxLengthForKeys` since it cannot make assumptions about the schema version?

Reply:
Yeah we'll have to decide where to turn this on, we likely can't change AddDefaultIdentity since existing apps are calling that.  I think this will be more clear once we start making breaking changes, the new APIs we add that require any new schema changes will likely be explicitly setting the version schema